pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go

```diff
@@ -111,8 +118,20 @@ func (c *connectivityCheckController) Sync(ctx context.Context, syncContext fact
 		return err
 	}
 
-	// TODO remove the following call in in 4.8
-	// re-create crd if deleted during an upgrade from 4.6
+	// do nothing while an upgrade is in progress
+	clusterVersion, err := c.clusterVersionLister.Get("version")
+	if err != nil {
+		return err
+	}
+	desired := clusterVersion.Status.Desired.Version
+	history := clusterVersion.Status.History
+	// upgrade is in progress if there is no history, or the latest history entry matches the desired version and is not completed
+	if len(history) == 0 || !(history[0].Version == desired && history[0].State != configv1.CompletedUpdate) {
```

Comment:
Are you sure this logic is correct? I expected (and have easier time reading)

```go
if len(history) == 0{
  klog.V(1).Infof("ConnectivityCheckController is waiting for transition to first desired version (%s) to be completed.", desired)
  return nil
}
if history[0].Version != desired{
  klog.V(1).Infof("ConnectivityCheckController is waiting for transition to desired version (%s) to be started.", desired)
  return nil
}

// this one looks different from what you wrote above....
if history[0].State != configv1.CompletedUpdate{
  klog.V(1).Infof("ConnectivityCheckController is waiting for transition to desired version (%s) to be completed.", desired)
  return nil
}

```

Reply:
Yes, I had a `==` that should of been a `!=` there. I'll split it out as per your suggestion.

Reply:
Updated.